helpers/network-recorder.js

```diff
@@ -21,14 +21,20 @@ const NetworkManager = require('./web-inspector').NetworkManager;
 const REQUEST_FINISHED = NetworkManager.EventTypes.RequestFinished;
 
 class NetworkRecorder {
-  constructor(recordArray) {
+  constructor(recordArray, requestListener) {
     this._records = recordArray;
 
+    this._listener = requestListener;
+
     this.networkManager = NetworkManager.createWithFakeTarget();
 
     // TODO(bckenny): loadingFailed calls are not recorded in REQUEST_FINISHED.
     this.networkManager.addEventListener(REQUEST_FINISHED, request => {
       this._records.push(request);
+
+      if (this._listener) {
```

Comment:
I think we sorted this one out today. We look in previous net reqs for the manifest, if its not there then we requestAppBanner and wait for it then. That will satisfy all cases.